pyaedt/emit.py

```diff
@@ -117,6 +196,266 @@ def __init__(
+        #   raise ValueError("This version of AEDT is unsupported for PYAEDT.")
+    @pyaedt_function_handler() 
     def __enter__(self):
         return self
+    @pyaedt_function_handler()
+    def analyze(self):
+        """
+        Analyze the Active Design.
+
+        Parameters
+        ----------
+
+        Returns
+        -------
+        :class:`pyaedt.modules.Revision`
+            Revision object.
+
+        References
+        ----------
+        >>> aedtapp.analyze()
+
+        """
+        design = self.odesktop.GetActiveProject().GetActiveDesign()
+        if(not self.curr_design == design.getRevision()):
+            design.AddResult()
+            self.location = self.oproject.GetPath()
+            self.revisions_list.append(Revision(self))
+            self.curr_design = design.getRevision()
+            print("checkpoint - revision generated successfully")
+            return self.revisions_list[-1]
+
+    @pyaedt_function_handler()
+    def load_revision(self, path) :
+        """
+        Load a specific revision.
+
+        Parameters
+        ----------
+        path:
+            Path to a .aedtresult file
+
+        Returns
+        -------
+        :class:`NoneType`
+
+        References
+        ----------
+        >>> aedtapp.load_revision(path)
+
+        """
+        self._emit_api.load_result(path)
+        self._result_loaded = True
+    def result_mode_error():
+        """ prints error message"""
+        print("This function is inaccessible when the Emit object has no revisions")
+    @pyaedt_function_handler()
+    def interaction_domain():
+        """
+        Return a generic interaction domain.
+
+        Parameters
+        ----------
+       
+        Returns
+        -------
+        :class:`InteractionDomain`
+
+        References
+        ----------
+        >>> Emit.interaction_domain()
+        """
+        interaction_domain = EmitApiPython.InteractionDomain()
+        return interaction_domain
+    @pyaedt_function_handler()
+    def result_type(mode =''): 
+        """
+        Return a result_type object or a specific type based on input.
+
+        Parameters
+        ----------
+        mode:
+            Type of result_type object, options include: "emi", "desense", "sensitivity". Default is ''.
+
+        Returns
+        -------
+        :class:`result_type`
+
+        References
+        ----------
+        >>> Emit.result_type("emi")
+
+        """
+        result_t = EmitApiPython.result_type
+        if mode == 'emi':
+            result_t = EmitApiPython.result_type.emi
+        elif mode == 'desense':
+             result_t = EmitApiPython.result_type.desense
+        elif mode == 'sensitivity':
+             result_t = EmitApiPython.result_type.sensitivity 
+        return result_t
+    @pyaedt_function_handler()
+    def tx_rx_mode(mode = ''): 
+        """
+        Return tx_rx_mode object or a specific mode based on input.
```

Comment:
```suggestion
        Return the ``tx_rx_mode`` object or a specific mode based on an input.
```

Reply:
Done! However, it should be noted that in a later commit - the mode parameter is removed, and a generic tx_rx_mode object is returned.